great_tables/_utils_nanoplots.py

```diff
@@ -935,15 +932,8 @@ def _generate_nanoplot(
     # Create normalized (and inverted for SVG) data `x` and `y` values for the
     # plot area; these are named `data_x_points` and `data_y_points`
     #
-
-    for i in range(len(y_proportions)):
-        y_proportions[i] = safe_y_d + ((1 - y_proportions[i]) * data_y_height)
-
-    for i in range(len(x_proportions)):
-        x_proportions[i] = (data_x_width * x_proportions[i]) + safe_x_d
-
-    data_y_points = y_proportions
-    data_x_points = x_proportions
+    data_y_points = tuple(safe_y_d + ((1 - y_p) * data_y_height) for y_p in y_proportions)
+    data_x_points = tuple((data_x_width * x_p) + safe_x_d for x_p in x_proportions)
```

Comment:
It looks like x_proportions and y_proportions are mutated often in the original code. Can we restore this bit of code back?

Even if the tests pass, I want to make sure we do a bigger sweep first over the logic in this function, to be sure its defining / mutating / redefining things like `y_proportions` doesn't produce surprising edge cases (that might not be tested)

Reply:
Thanks for taking the time to look into how these are used. Let's just get this PR merged, since it seems like it shouldn't change things!